Extension/src/LanguageServer/extension.ts

```diff
@@ -331,25 +331,32 @@ function realActivation(): void {
 
     vcpkgDbPromise = initVcpkgDatabase();
 
-    PlatformInformation.GetPlatformInformation().then(info => {
+    PlatformInformation.GetPlatformInformation().then(async info => {
```

Comment:
Can `PlatformInformation.GetPlatformInformation().then(async info =>` be refactored so it doesn't need to use `.then`? For example, use `await` on `PlatformInformation.GetPlatformInformation()`.

Or is this one of those cases where `.then` is required to achieve the correct behavior?

Reply:
This is intentional, to chain functionality without waiting for it to complete.  It could take up to 6 seconds for the `getExperimentationServiceAsync` call to complete, and we don't want to stall extension activation until that is done.  (Or, we don't need to yet.  If we ever add an experiment variable that we need to read prior to activation to set things up properly, this would need to be awaited, and we'd just have to accept the delay.)